eclass/java-pkg-simple.eclass

```diff
@@ -335,19 +412,70 @@ java-pkg-simple_prepend_resources() {
 	done
 }
 
-# @FUNCTION: java-pkg-simple_src_compile
+# @FUNCTION: java-pkg-simple_call_ejavadoc
+# @USAGE: java-pkg-simple_call_ejavadoc
 # @DESCRIPTION:
-# src_compile for simple bare source java packages. Finds all *.java
-# sources in ${JAVA_SRC_DIR}, compiles them with the classpath
-# calculated from ${JAVA_GENTOO_CLASSPATH}, and packages the resulting
-# classes to a single ${JAVA_JAR_FILENAME}. If the file
+# Generates documentation
+java-pkg-simple_call_ejavadoc() {
+	local sources=docsources.lst apidoc=target/api
+	local modulesourcepath
+
+	local classpath=""
+	java-pkg-simple_getclasspath
+	local dependency
+	for dependency in ${JAVADOC_CLASSPATH}; do
+		classpath="${classpath}:$(java-pkg_getjars \
+			--build-only \
+			--with-dependencies \
+			${dependency})"
+	done
+
+	if ! [[ "${JAVADOC_SRC_DIRS}" ]]; then
+		local -a JAVADOC_SRC_DIRS
+		JAVADOC_SRC_DIRS=( "${JAVA_INTERNAL_SRC_DIRS[@]}" )
+	fi
+	# gather sources
+	# if target < 9, we need to compile module-info.java separately
+	# as this feature is not supported before Java 9
+	local target="$(java-pkg_get-target)"
+	if [[ ${target#1.} -lt 9 ]]; then
```

Comment:
Could probably be simplified to something like

```bash
[[ ${target#1.} -lt 9 ]] && find_args+=(! -name module-info.java)
```

Also, missing `die` on find invocations.

Reply:
Good idea. Same for the `ejavac` calls. But maybe that should be done in a separate PR, as this was not introduced here and this PR is already quite big. What do you think?

Reply:
> Good idea. Same for the `ejavac`

usually we do not `|| die` after ejavac, ejunit, einfo, emake ...

Reply:
I meant the first part, the simplification. That can be used for `ejavac`, too. Here:
```bash
    if [[ -z ${moduleinfo} ]] || [[ ${target#1.} -lt 9 ]]; then
        ejavac -d ${classes} -encoding ${JAVA_ENCODING}\
            ${classpath:+-classpath ${classpath}} ${JAVAC_ARGS} @${sources}
    else
        ejavac -d ${classes} "${modulesourcepath[@]}" -encoding ${JAVA_ENCODING}\
            ${classpath:+--module-path ${classpath}} --module-version ${PV}\
            ${JAVAC_ARGS} @${sources}
    fi
```
Use an `ejavac_args` array and only a single call to `ejavac` after the if.